pallets/identity-management-mock/src/lib.rs

```diff
@@ -318,7 +321,21 @@ pub mod pallet {
 			let decrypted_identitty = Self::decrypt_with_tee_shielding_key(&encrypted_identity)?;
 			let identity = Identity::decode(&mut decrypted_identitty.as_slice())
 				.map_err(|_| Error::<T>::WrongDecodedType)?;
-
+			if let Identity::Substrate { network, address } = identity {
+				// see all the address prefix:
+				// https://github.com/paritytech/ss58-registry/blob/main/ss58-registry.json
+				let ss58_prefix = T::SS58Prefix::get();
+				if (network == SubstrateNetwork::Litentry && ss58_prefix == 31) ||
+					(network == SubstrateNetwork::Litmus && ss58_prefix == 131)
+				{
+					let address_raw: [u8; 32] = who
+						.encode()
+						.try_into()
+						.map_err(|_| DispatchError::Other("invalid account id"))?;
+					let user_address: Address32 = address_raw.into();
+					ensure!(user_address != address, Error::<T>::IdentityShouldBeDisallowed);
+				}
+			}
```

Comment:
Looks good, thanks. Possible to have a unit test for the mock pallet? 

Btw maybe let's add another network type `SubstrateNetwork::LitentryRococo` -> will be used in testing

Reply:
where can I find the ss58 prefix of `LitentryRococo`?

Reply:
https://github.com/litentry/litentry-parachain/blob/2d25dd83fed8fdfbfa67eab922b191d5bcce936d/runtime/rococo/src/lib.rs#L170

So it's 131